src/operator/trace.rs

```diff
@@ -374,31 +375,26 @@ where
         self.time.advance(scope + 1);
     }
 
-    fn summary(&self, summary: &mut String) {
-        writeln!(
-            summary,
-            "size: {}",
-            self.trace
-                .as_ref()
-                .map(|trace| trace.num_entries_deep())
-                .unwrap_or(0)
-        )
-        .unwrap();
+    fn metadata(&self, meta: &mut OperatorMeta) {
+        let total_size = self
+            .trace
+            .as_ref()
+            .map(|trace| trace.num_entries_deep())
+            .unwrap_or(0);
 
         let bytes = self
             .trace
             .as_ref()
             .map(|trace| trace.size_of())
             .unwrap_or_default();
-        writeln!(
-            summary,
-            "allocated: {}, used: {}, allocations: {}, shared: {}",
-            HumanBytes::from(bytes.total_bytes()),
-            HumanBytes::from(bytes.used_bytes()),
-            bytes.distinct_allocations(),
-            HumanBytes::from(bytes.shared_bytes()),
-        )
-        .unwrap();
+
+        meta.extend(metadata! {
+            "total size" => total_size,
+            "allocated bytes" => MetaItem::bytes(bytes.total_bytes()),
```

Comment:
It might be worth having a method to convert `TotalSize` to metadata.  If we also support nested metadata objects, all `TotalSize` elements could be grouped under one attribute.